packages/fxa-settings/src/components/Settings/UnitRowRecoveryKey/index.tsx

```diff
@@ -2,116 +2,174 @@
+  }, [account, hideModal, alertBar, ftlMsgResolver]);
 
-  const localizedRefreshRkText = l10n.getString(
+  const localizedRefreshRkText = ftlMsgResolver.getMsg(
     'rk-refresh-key-1',
-    null,
     'Refresh account recovery key'
   );
 
+  const localizedDeleteRKIconButton = ftlMsgResolver.getMsg(
+    'unit-row-recovery-key-delete-icon-button-title',
+    'Delete recovery key'
+  );
+
   return (
     <UnitRow
-      header={l10n.getString('rk-header-1', null, 'Account recovery key')}
+      header={ftlMsgResolver.getMsg('rk-header-1', 'Account recovery key')}
       headerId="recovery-key"
       prefixDataTestId="recovery-key"
       headerValueClassName={recoveryKey ? 'text-green-800' : ''}
       headerValue={
         recoveryKey
-          ? l10n.getString('rk-enabled', null, 'Enabled')
-          : l10n.getString('rk-not-set', null, 'Not Set')
+          ? ftlMsgResolver.getMsg('rk-enabled', 'Enabled')
+          : ftlMsgResolver.getMsg('rk-not-set', 'Not Set')
+      }
+      // TODO Remove condition in FXA-7419 and only keep v2
+      route={
+        showRecoveryKeyV2
+          ? `${HomePath}/account_recovery`
+          : recoveryKey
+          ? undefined
+          : `${HomePath}/account_recovery`
+      }
+      // Remove this attribute when v1 phased out in FXA-7419
+      revealModal={
+        showRecoveryKeyV2 ? undefined : recoveryKey ? revealModal : undefined
       }
-      route={recoveryKey ? undefined : `${HomePath}/account_recovery`}
-      revealModal={recoveryKey ? revealModal : undefined}
+      // TODO Remove condition in FXA-7419 and only keep v2
       ctaText={
-        recoveryKey
-          ? l10n.getString('rk-action-remove', null, 'Remove')
-          : l10n.getString('rk-action-create', null, 'Create')
+        showRecoveryKeyV2
+          ? recoveryKey
+            ? ftlMsgResolver.getMsg('rk-action-change-button', 'Change')
+            : ftlMsgResolver.getMsg('rk-action-create', 'Create')
+          : recoveryKey
+          ? ftlMsgResolver.getMsg('rk-action-remove', 'Remove')
+          : ftlMsgResolver.getMsg('rk-action-create', 'Create')
       }
       disabled={!account.hasPassword}
-      disabledReason={l10n.getString(
+      disabledReason={ftlMsgResolver.getMsg(
         'security-set-password',
-        null,
         'Set a password to sync and use certain account security features.'
       )}
       alertBarRevealed
       headerContent={
-        <ButtonIconReload
-          title={localizedRefreshRkText}
-          classNames="mobileLandscape:hidden ltr:ml-1 rtl:mr-1"
-          disabled={account.loading}
-          onClick={() => account.refresh('recovery')}
-        />
+        // TODO Remove condition in FXA-7419 and only keep v2
+        showRecoveryKeyV2 ? (
+          <ButtonIconTrash
+            title={localizedDeleteRKIconButton}
+            classNames={`mobileLandscape:hidden ms-1 h-11 ${
+              recoveryKey ? '' : 'invisible'
+            }`}
```

Comment:
if we use the `classnames` package, we _might_ be able to do something like this (although maybe debatable as to whether it's a great improvement):

```js
        classNames={classNames('mobileLandscape:hidden ms-1 h-11', {
          'invisible': !recoveryKey,
        })}
```

Reply:
Opted to conditionally load the button icon instead 👍